### ISSUE TITLE
Implement proper signal handling and exiting

### ISSUE THREAD

**Issue by kragniz** [sails-simulator/sailsd#2]
C-c currently just calls `exit(1)`. This should be improved to cleanly close sockets, write to files and things.

**Comment by Danyc0** [comment]
The program already seems catch the SIGINT here:
https://github.com/sails-simulator/sailsd/blob/master/sailsd.c#L382-L386
So is it just that the code for closing the sockets needs to happen here?

**Comment by kragniz** [comment]
Yes, it's a question of getting the file handler for the socket (https://github.com/sails-simulator/sailsd/blob/master/sailsd.c#L435) into that scope, then dealing with it.

**Comment by Danyc0** [comment]
Looks like the only way to do that is to put the socket descriptor in the module scope, right? because there's no other way to get a variable into a signal handlers scope

**Comment by kragniz** [comment]
Yup, something like that would need to be done.
